contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -2174,3 +2174,21 @@ patindex_ai_collations(PG_FUNCTION_ARGS)
 
 	PG_RETURN_INT32(result);
 }
+
+
+void
+assign_psql_logical_babelfish_db_name(const char *newval, void *extra)
+{
+	/* 
+	 * No need to tell babelfishpg_common that logical db name has been set if dialect is TSQL
+	 * because the psql_logical_babelfish_db_name is to be used for PG dialect only
+	 */
+	if (sql_dialect == SQL_DIALECT_PG)
+	{
+		/* Initialise collation callbacks */
+		init_and_check_collation_callbacks();
+
+		/* let babelfishpg_common know that psql_logical_babelfish_db_name has been updated */
+		(*collation_callbacks_ptr->set_logical_db_name_cache) (newval);
```

Comment:
Given this logic, shall we check for valid values of newval?

Reply:
IMO, that's redundant as even if the logical db name is incorrect, we won't find any relations, functions, view, procedures etc for that incorrect/invalid logical db